### PR TITLE
Allow appliance vm attach multiple system network

### PR DESCRIPTION
**Pull request by MatheMatrix**
For zstackio/issues#4179

### CODE REVIEW

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java [hunk 1]
```diff
@@ -2916,23 +2917,56 @@ public void run(final SyncTaskChain chain) {
                     return;
                 }
 
+                FlowChain fchain = FlowChainBuilder.newSimpleFlowChain();
+                fchain.setName(String.format("detach-l3-network-to-vm-%s", msg.getVmInstanceUuid()));
+                fchain.then(new NoRollbackFlow() {
```
Comment (zxwing) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: 用__name__给flow加个名字

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java [hunk 2]
```diff
@@ -2916,23 +2917,56 @@ public void run(final SyncTaskChain chain) {
...
+                            @Override
+                            public void fail(ErrorCode errorCode) {
+                                trigger.fail(errorCode);
+                            }
+                        });
+                    }
+                }).then(new NoRollbackFlow() {
                     @Override
-                    public void success() {
+                    public void run(FlowTrigger trigger, Map data) {
```
Comment (zxwing) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: 用__name__给flow加个名字

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java [hunk 3]
```diff
@@ -2916,23 +2917,56 @@ public void run(final SyncTaskChain chain) {
...
+                        detachNic(msg.getVmNicUuid(), new Completion(trigger) {
+                            @Override
+                            public void success() {
+                                self = dbf.reload(self);
```
Comment (zxwing) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: 不必要的reload

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java [hunk 4]
```diff
@@ -3531,20 +3569,60 @@ public void fail(ErrorCode errorCode) {
...
+                });
+            }
+        }).then(new NoRollbackFlow() {
+            @Override
+            public void run(FlowTrigger trigger, Map data) {
+                afterAttachNic((VmNicInventory) data.get(vmNicInvKey), new Completion(trigger) {
+                    @Override
+                    public void success() {
+                        trigger.next();
+                        return;
```
Comment (zxwing) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: 不必要的return

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java (hunk 1 above)
Comment (MatheMatrix) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: done

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java (hunk 2 above)
Comment (MatheMatrix) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: done

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java (hunk 3 above)
Comment (MatheMatrix) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: done

File: compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java (hunk 4 above)
Comment (MatheMatrix) [inline compute/src/main/java/org/zstack/compute/vm/VmInstanceBase.java]: done

**Pull request merged**